### DISCUSSION TITLE
How to trigger validation initially?

### DISCUSSION THREAD

**Discussion by anhquande** [react-hook-form/react-hook-form#2573] [Q&A] [answered]
Hello,

In my project, i have a form in a Dialog. When dialog opens, data are loaded and placed in the form.
I want to manually trigger the form validation only when form is loaded or submitted. It means validation will not be executed during field onChange.

I realize that trigger() does nothing in the first and second call. That is why i have to use the counter to stop trigger() after the 3rd call.

Here are my code

    function useValidateFormInitially({open, form}) {
    
      const [count, setCount] = useState(0);
      const {trigger} = form;
    
      useEffect(()=>{
        if (open && count <3){
          trigger();
          setCount(prev => prev+1);
        }
      }, [open, trigger, count])
    
    }

Did i do anything wrong? or there is a bug on trigger() .

Thanks

**Comment by bluebill1049** [comment, marked as answer]
u have a csb?

**Reply by anhquande** [reply]
sorry, what is csb?

**Reply by bluebill1049** [reply]
codesandbox

**Reply by anhquande** [reply]
here it is: https://codesandbox.io/s/nice-dirac-ul2w9

**Reply by bluebill1049** [reply]
please fix those warning in the codesandbox, I suspect this is due to the transition with the modal.

**Reply by anhquande** [reply]
i did it

**Reply by bluebill1049** [reply]
<img width="1259" alt="Screen Shot 2020-08-12 at 8 36 14 pm" src="https://user-images.githubusercontent.com/10513364/90006024-7a897a00-dcdb-11ea-9faa-a8fd8e1642b3.png">
see those warnings.

**Reply by anhquande** [reply]
it is now warning free

**Reply by bluebill1049** [reply]
This is a problem due to all the inputs are not mounted yet, so which means nothing is registered (due to the animation), you probably can avoid this problem:

1. if you trigger within the form inputs
2. some sort of callback when inputs are mounted.

**Reply by anhquande** [reply]
My solution is to create a TriggerValidation component. This component will be placed at the end of the form, to be sure that it will be rendered at the latest. And it works! 

Here are my changes. https://codesandbox.io/s/nostalgic-lewin-1d5xn

**Reply by bluebill1049** [reply]
Awesome! thanks for sharing solution @anhquande
